.github/workflows/examples.yml

```diff
@@ -1,88 +1,79 @@
 name: Examples
 
 on:
-  push:
-    paths:
-      - "**.py"
-      - "**.yml"
-  pull_request:
-
-  workflow_dispatch:
+  workflow_call:
 
 jobs:
   examples:
+    name: Run Examples
+    runs-on: ubuntu-20.04
+    timeout-minutes: 30
     strategy:
       fail-fast: false
       matrix:
-        include:
-          - python-version: "3.11"
-            exasol-tag: latest-7.1
-            exasol-port: 8563
-
-    name: "Build with Python ${{ matrix.python-version }} and Exasol ${{ matrix.exasol-tag }}"
-
-    runs-on: ubuntu-20.04
-    timeout-minutes: 15
-
-    services:
-      exasol:
-        image: exasol/docker-db:${{ matrix.exasol-tag }}
-        ports:
-          - 8563:${{ matrix.exasol-port }}
-        options: --privileged
-
+        python-version: ["3.11"]
     steps:
-      - name: "Checkout repo (Python ${{ matrix.python-version }} Exasol ${{ matrix.exasol-tag }})"
+      - name: "Checkout repo"
         uses: actions/checkout@v2
 
-      - name: "Setup Python (Python ${{ matrix.python-version }} Exasol ${{ matrix.exasol-tag }})"
-        uses: actions/setup-python@v2
+      - name: "Setup Python & Poetry Environment"
+        uses: exasol/python-toolbox/.github/actions/python-environment@0.7.0
         with:
           python-version: ${{ matrix.python-version }}
 
-      - name: "Install dependencies (Python ${{ matrix.python-version }} Exasol ${{ matrix.exasol-tag }})"
-        run: pip install -e .[pandas,ujson,rapidjson,orjson,examples]
+      - name: "Install all extras"
+        run: poetry install --all-extras
+
+      - name: "Start test db"
+        run: >
+          poetry run itde spawn-test-environment --environment-name test 
+          --database-port-forward 8563 --bucketfs-port-forward 2580 
+          --docker-db-image-version 7.1.17 --db-mem-size 4GB
 
-      - name: "Wait for Exasol container (Python ${{ matrix.python-version }} Exasol ${{ matrix.exasol-tag }})"
-        run: python examples/_wait_container.py
+      - name: "Wait for Exasol container (UDF extraction)"
+        run: poetry run python examples/_wait_container.py
+
+      - name: "Run examples"
+        run: |
+          poetry run python examples/a00_prepare.py
```

Comment:
Can you list all files in the directory and run them all, instead of listing them?

Reply:
Don't wan't to change the behavior of this workflow to much, also the order of execution is important to some tests.